lib/kernel-manager.coffee

```diff
@@ -149,8 +149,8 @@ module.exports = KernelManager =
 
         kernelSpec.grammarLanguage = grammarLanguage
 
-        ConfigManager.writeConfigFile (filepath, config) =>
-            kernel = new Kernel kernelSpec, grammar, config, filepath
+        ConfigManager.writeConfigFile (filepath, config, onlyConnect = false) =>
```

Comment:
I think the functionality checking for `customKernelConnectionPath` belongs to `KernelManager`.

`ConfigManager.writeConfigFile` should only be called when a config file is written.